### PR TITLE
Package Info Api

### LINKED ISSUE #5: getPackageInfo() method
Create getPackageInfo() method

### PR DESCRIPTION
**Pull request by BaderBC**
Closes #5
Implements docker for testing

- package info API done
- docker config for tests
- package test added

### CODE REVIEW

File: src/api/package.go
Comment (sewe2000) [inline src/api/package.go:1]: I think the function 'getPackageInfo' should be renamed to 'GetPackageInfo' in order to export this function (so it could be used in other go files such as main.go). I wonder if the 'SetField' method is really needed (do you want to implement some kind of validation?). There's also a big problem regarding the 'Get' method because I presume it only works with the English locale. For example if a user has polish locale, then this function won't work because there's no field 'Licencje' in struct 'Package'. We need to discuss this issue more thoroughly. Overall the code is good

**Review by cooligus — COMMENTED** [review]
IMO test is currently like a small boy shouting at you "add me to your class group!"

**Comment by BaderBC** [comment]
> IMO test is currently like a small boy shouting at you "add me to your class group!"

I think that we should use test driven architecture or at least something similar.
This project, especially API part is just library, so it is hard to develop code without any tests.

File: src/api/package.go
Comment (BaderBC) [reply src/api/package.go:1]: getPackageInfo is private on purpose, because it returns package info as (string, error). I don't think string containing output of "pacman -Qi package" command will be useful in any other file.

setField is method which I created to avoid putting all logic into one long function. Maybe I'm wrong, but I think it makes code cleaner and more readable. (body of setField could be not really meaningful so I separate this into function called setField which explains what that block of code does)

Speaking about the biggest problem that you fortunately found, I find out how to resolve. Here's how: "LANG=C pacman -Qi package". I will fix this one in next commit.

**Review by cooligus — COMMENTED** [review]
So are you going to create test in this pull request or maybe in another?

**Comment by BaderBC** [comment]
Just added GitHub actions CI for testing

File: .github/workflows/test.yml
Comment (maxidragon) [inline .github/workflows/test.yml:1]: Please change name to Test

**Review by maxidragon — CHANGES_REQUESTED** [review]

**Review by maxidragon — APPROVED** [review]
LGTM, but please wait for someone else to approve before merging

**Review by cooligus — APPROVED** [review]

**Review by sewe2000 — APPROVED** [review]

File: src/api/package.go
Comment (BaderBC) [reply src/api/package.go:1]: setField make code cleaner, because it contain additional logic beyond just assigning value to key

**Pull request merged**
